### PR TITLE
Build retrieval improvements

### PR DESCRIPTION
**Pull request by vitorguidi**
### Motivation

Small changes for the build retrieval metric, as requested per Chrome folks after initially using the feature:

* Adding the time for listing fuzz targets as a step
* Added the total retrieval duration
* Tracking metric as minutes, for readability

Part of #4271

### CODE REVIEW

**Review by paulsemel — APPROVED** [review]
Thanks for working on this!

File: src/clusterfuzz/_internal/build_management/build_manager.py [hunk 1]
```diff
@@ -528,7 +529,17 @@ def _unpack_build(self,
         if not unpack_everything:
           # We will never unpack the full build so we need to get the targets
           # from the build archive.
+          list_fuzz_target_start_time = time.time()
           self.fuzz_targets = list(build.list_fuzz_targets())
+          # In minutes, as per metric definition.
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/build_management/build_manager.py:534]: You mean like doing a `with ....:` ?

**Review by jonathanmetzman — COMMENTED** [review]
lgtm

File: src/clusterfuzz/_internal/build_management/build_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/build_management/build_manager.py:534]: This sounds good to me!

**Review by jonathanmetzman — APPROVED** [review]

File: src/clusterfuzz/_internal/build_management/build_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vitorguidi) [reply src/clusterfuzz/_internal/build_management/build_manager.py:537]: Now that I actually tried to write the code, a context manager is not a good idea. A simple function suffices to declutter the code.

File: src/clusterfuzz/_internal/build_management/build_manager.py [hunk 2]
```diff
@@ -522,7 +526,11 @@ def _unpack_build(self,
         if not self._unpack_everything:
           # We will never unpack the full build so we need to get the targets
           # from the build archive.
+          list_fuzz_target_start_time = time.time()
           self._fuzz_targets = list(build.list_fuzz_targets())
+          _emit_job_build_retrieval_metric(list_fuzz_target_start_time,
```
Comment (paulsemel) [inline src/clusterfuzz/_internal/build_management/build_manager.py:531]: You also need to emit this metric in the `fuzz_targets` property method. Meant to comment there, but it looks like I can't.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Pull request merged**